packages/dev/core/src/Materials/Node/Blocks/Fragment/fragmentOutputBlock.ts

```diff
@@ -87,9 +94,34 @@ export class FragmentOutputBlock extends NodeMaterialBlock {
         return this._inputs[2];
     }
 
+    /**
+     * Gets the depth input component - only available if prepass outputs
+     * are turned on
+     */
+    public get depth(): NodeMaterialConnectionPoint {
+        return this._inputs[3];
+    }
+
+    /**
+     * Gets the depth input component - only available if prepass outputs
+     * are turned on
+     */
+    public get worldPosition(): NodeMaterialConnectionPoint {
+        return this._inputs[4];
+    }
+
+    /**
+     * Gets the depth input component - only available if prepass outputs
+     * are turned on
+     */
+    public get worldNormal(): NodeMaterialConnectionPoint {
+        return this._inputs[5];
+    }
+
     public prepareDefines(mesh: AbstractMesh, nodeMaterial: NodeMaterial, defines: NodeMaterialDefines) {
         defines.setValue(this._linearDefineName, this.convertToLinearSpace, true);
         defines.setValue(this._gammaDefineName, this.convertToGammaSpace, true);
+        defines.markAsPrePassDirty();

```

Comment:
It was necessary because the flag `isPrePassCapable` was depending on user input, and thus on `NodeMaterial`.

As you pointed out in your comment, `isPrePassCapable` should not be necessary anymore, making this call useless